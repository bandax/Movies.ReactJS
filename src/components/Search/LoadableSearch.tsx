import * as React from 'react';
import Loadable from 'react-loadable';
import Loading from '../Loading/Loading';

const LoadableSearch = Loadable({
  loader: () => import('./Search'),
  loading: Loading,
});

const SearchSplit: React.FunctionComponent = () => {
  return <LoadableSearch />;
};

export default SearchSplit;
