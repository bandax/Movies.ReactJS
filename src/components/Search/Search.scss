.search{
    text-align: left;     
    text-transform: uppercase;   
    padding: 100px 100px;  
}

.find-label {
    font-size: 30px;
    color: #FFFFFF;
}

.search-box {
    background-color: rgba(10, 10, 10, 0.8);  
    font-size: 20px;
    margin-right: 20px;
    padding: 10px 25px;    
    border-radius: 8px;  
    width: 70%; 
    border-color: #232323;
}

.btn-search {
    background-color: #F65261;
    color: white;
    font-size: 20px;
    padding: 10px 50px;    
    border-radius: 5px;
    text-transform: uppercase; 
}
