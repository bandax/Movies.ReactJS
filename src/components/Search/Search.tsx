import * as React from 'react';
import './Search.scss';

const Search: React.FunctionComponent = ()  => (   
    <div className="search">
        <span className="find-label">Find your movie</span>
        <br />
        <input className="search-box" placeholder="What do you want to watch?" />
        <button className="btn btn-search">Search</button>
    </div>   
);

export { Search };
