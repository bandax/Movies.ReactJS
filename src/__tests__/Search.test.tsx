import * as React from 'react';
import { Search } from '../components/Search/Search';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';

describe('<Search />', () => {
  test('render a Search Component', async () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('type a text to search', async () => {
    const { getByRole } = render(<Search />);
    const input = getByRole('textbox');
    userEvent.type(input, 'Fifty');
    expect(input).toHaveValue('Fifty');
  });

  test('click search button to redirect search page with text included', async () => {
    const { getByText, getByRole } = render(<Search />);
    const input = getByRole('textbox');
    const link = getByText('Search');
    userEvent.type(input, 'Fifty');
    expect(input).toHaveValue('Fifty');
    expect(link.closest('a')).toHaveAttribute('href', '#/search/Fifty');
  });
});
