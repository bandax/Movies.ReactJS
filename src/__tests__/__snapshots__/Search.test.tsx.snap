// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> render a Search Component 1`] = `
<DocumentFragment>
  <div
    class="search col-10"
  >
    <span
      class="find-label"
    >
      Find your movie
    </span>
    <br />
    <input
      class="search-box"
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <a
      class="btn btn-search"
      href="#/search/"
    >
      Search
    </a>
  </div>
</DocumentFragment>
`;
